scalyr_agent/builtin_monitors/thread_pool.py

```diff
@@ -20,23 +20,38 @@
 from concurrent.futures import ThreadPoolExecutor
 
 from scalyr_agent import scalyr_logging
+import time
 
 global_log = scalyr_logging.getLogger(__name__)
 
 # A MixIn class used for adding a thread poll processing to a BaseServer (i.e. SyslogTCPServer, SyslogUDPServer)
 class ExecutorMixIn:
     def __init__(self, global_config):
-        # Using 4 threads for processing requests, because of GIL and CPU bound tasks higher number would not help process the data faster.
-        reading_threads = 4
+        # Using 8 threads for processing requests, because of GIL and CPU bound tasks higher number would not help process the data faster.
+        # The reason for multiple threads is to avoid deadlock in an unexpected situation before waiting for previous data
+        # on a single request being processed fails and it's not caught do to a bug.
+        processing_threads = 8
         # Let the ThreadPoolExecutor decide how many threads to use based the numbre of logical CPUs
-        processing_threads = None
+        reading_threads = None
 
         if global_config:
             reading_threads = global_config.syslog_socket_thread_count
             processing_threads = global_config.syslog_processing_thread_count
+            self._shutdown_grace_period = global_config.syslog_monitors_shutdown_grace_period
+        else:
```

Comment:
I assume we end in this else branch inside tests?

Reply:
No, inside the IF branch, there's a MockConfig being passed.
I've created a test to run it without global_config.